dist/zip-fs-full.js

```diff
@@ -7127,22 +7128,35 @@
 			this.preventHeadRequest = options.preventHeadRequest;
 			this.useRangeHeader = options.useRangeHeader;
 			this.forceRangeRequests = options.forceRangeRequests;
+			this.useGetForRange = options.useGetForRange;
 			this.options = Object.assign({}, options);
 			delete this.options.preventHeadRequest;
 			delete this.options.useRangeHeader;
 			delete this.options.forceRangeRequests;
+			delete this.options.getForRange;
 			delete this.options.useXHR;
 		}
 
 		async init() {
 			super.init();
-			if (isHttpFamily(this.url) && !this.preventHeadRequest) {
-				const response = await sendFetchRequest(HTTP_METHOD_HEAD, this.url, this.options);
-				this.size = Number(response.headers.get(HTTP_HEADER_CONTENT_LENGTH));
-				if (!this.forceRangeRequests && this.useRangeHeader && response.headers.get(HTTP_HEADER_ACCEPT_RANGES) != HTTP_RANGE_UNIT) {
-					throw new Error(ERR_HTTP_RANGE);
-				} else if (this.size === undefined) {
-					await getFetchData(this, this.options);
+			if (isHttpFamily(this.url) && (!this.preventHeadRequest || this.useGetForRange)) {
+				if (this.useGetForRange) {
+					const response = await sendFetchRequest(HTTP_METHOD_GET, this.url, this.options, { range: "bytes=0-0"});
```

Comment:
You must replace `{ range: "bytes=0-0"}` with ``[[HTTP_HEADER_RANGE, HTTP_RANGE_UNIT + "=0-0"]]`.